test/gtest/uct/test_ud.cc

```diff
@@ -120,14 +126,18 @@ class test_ud : public ud_base_test {
         EXPECT_EQ(value, ep->ep_id);
     }
 
+    unsigned no_creq_cnt(uct_ud_ep_t *ep) {
+        return (ep->flags & UCT_UD_EP_FLAG_CREQ_NOTSENT) ? 1 : 0;
+    }
+
     void validate_recv(uct_ud_ep_t *ep, unsigned value,
                        double timeout_sec=TEST_UD_TIMEOUT_IN_SEC) {
         ucs_time_t timeout = ucs_get_time() + ucs_time_from_sec(timeout_sec);
-        while ((ucs_frag_list_sn(&ep->rx.ooo_pkts) < value) &&
+        while ((ucs_frag_list_sn(&ep->rx.ooo_pkts) < value - no_creq_cnt(ep)) &&
```

Comment:
this test expects CREP to arrive, in case we actually sent the CREQ (actually, we count received CREPs rather than xmitted CREQs here)